launcher/ui/pages/instance/ManagedPackPage.cpp

```diff
@@ -338,6 +337,25 @@ void ModrinthManagedPackPage::suggestVersion()
     ManagedPackPage::suggestVersion();
 }
 
+/// @brief Called when the update task has completed.
+/// Internally handles the closing of the instance window if the update was successful and shows a message box.
+/// @param did_succeed Whether the update task was successful.
+void ManagedPackPage::onUpdateTaskCompleted(bool did_succeed) const
+{
+    // Close the window if the update was successful
+    if (did_succeed) {
+        if (m_instance_window != nullptr)
+            m_instance_window->close();
+
+        CustomMessageBox::selectable(nullptr, tr("Update Successful"), tr("The instance updated to pack version %1 successfully.").arg(m_inst->getManagedPackVersionName()), QMessageBox::Information)
+           ->show();
+    } else {
+        CustomMessageBox::selectable(nullptr, tr("Update Failed"), tr("The instance failed to update to pack version %1. Please check launcher logs for more information.").arg(m_inst->getManagedPackVersionName()), QMessageBox::Critical)
```

Comment:
Uh sorry I missed this before but I don't think it's a good idea to call getManagedPackVersionName as it might be the old version

Reply:
Hmm. I think it would be useful to have that data. In my testing, that did return the new version, but is there a better way?

Reply:
Not from what I know.
@TheKodeToad are you ok with this? Or can you propose an alternative?

Reply:
If the pack version is changed in a failed update, that is probably a bug, right?
So even if it does that right now... it should probably be changed to roll back changes

Reply:
But that is not related to this PR.
You're asking for a bug fix on a feature implementation.
In my opinion, the rollback thing can be done in a separate PR.

Reply:
If it's fixed this will be broken, that is the thing

Reply:
If it fixed then it should also fix the dialog

Reply:
I guess it doesn't really matter